### PR TITLE
feat: Todo application

### PR DESCRIPTION
**Pull request by richajituri1**
Explain why this change is being made: -
- Flutter application which displays Todo and allows user to view the Todo in ascending as well as descending order

### CODE REVIEW

**Review by KaustubhDoshaheen — CHANGES_REQUESTED** [review]

File: lib/clickable_todo_item.dart [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import 'package:flutter/material.dart';
+
+enum Priority { urgent, normal, low }
```
[the remark this answers is not in the document]
Comment (richajituri1) [reply lib/clickable_todo_item.dart]: Made the changes and pushed the code

File: lib/clickable_todo_item.dart [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+class CheckableTodoItem extends StatefulWidget {
+  const CheckableTodoItem(this.text, this.priority, {super.key});
```
[the remark this answers is not in the document]
Comment (richajituri1) [reply lib/clickable_todo_item.dart]: Made the changes, used named arguments

File: lib/todo_item.dart [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import 'package:flutter/material.dart';
+
+enum Priority { urgent, normal, low }
+
+class TodoItem extends StatelessWidget {
+  const TodoItem(this.text, this.priority, {super.key});
```
[the remark this answers is not in the document]
Comment (richajituri1) [reply lib/todo_item.dart]: Used named arguments

File: lib/keys.dart [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+  void _changeOrder() {
+    setState(() {
+      _order = _order == 'asc' ? 'desc' : 'asc';
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
```
[the remark this answers is not in the document]
Comment (KaustubhDoshaheen) [reply lib/keys.dart:63]: @richajituri1 @RichaJituri  Please do not mark comments resolved but Add Comment saying `Resolved`

File: lib/keys.dart [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  const Keys({super.key});
+
+  @override
+  State<Keys> createState() {
+    return _KeysState();
+  }
+}
+
+class _KeysState extends State<Keys> {
+  var _order = 'asc';
```
Comment (KaustubhDoshaheen) [inline lib/keys.dart]: shift this isolated string to constants

File: lib/keys.dart [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+  List<Todo> get _orderedTodos {
+    final sortedTodos = List.of(Constants.todos);
+    sortedTodos.sort((a, b) {
+      final bComesAfterA = a.text.compareTo(b.text);
+      return _order == 'asc' ? bComesAfterA : -bComesAfterA;
```
Comment (KaustubhDoshaheen) [inline lib/keys.dart]: same

File: lib/keys.dart [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    });
+    return sortedTodos;
+  }
+
+  void _changeOrder() {
+    setState(() {
+      _order = _order == 'asc' ? 'desc' : 'asc';
```
Comment (KaustubhDoshaheen) [inline lib/keys.dart]: same

File: lib/keys.dart [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    });
+  }
+
+  Widget _buildButton() {
+    return TextButton.icon(
+      onPressed: _changeOrder, 
+      icon: Icon(
+        _order == 'asc' ? Icons.arrow_downward : Icons.arrow_upward,
+        ),
+        label: Text('Sort ${_order == 'asc' ? 'Descendin' : 'Ascending'}'),
```
Comment (KaustubhDoshaheen) [inline lib/keys.dart]: same sift strings to constants and access that over here

File: lib/keys.dart [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+    setState(() {
+      _order = _order == 'asc' ? 'desc' : 'asc';
+    });
+  }
+
+  Widget _buildButton() {
+    return TextButton.icon(
+      onPressed: _changeOrder, 
+      icon: Icon(
+        _order == 'asc' ? Icons.arrow_downward : Icons.arrow_upward,
```
Comment (KaustubhDoshaheen) [inline lib/keys.dart]: same

File: lib/keys.dart [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+    return Column(
+      children: [
+        Align(
+          alignment: Alignment.centerRight,
+          child: _buildButton(),
+        ),
+        Expanded(
+          child: Column(
+            children: [
+              for (final todo in _orderedTodos)
```
Comment (KaustubhDoshaheen) [inline lib/keys.dart]: consider creating a function that performs this looping logic and return widget, or consider using list view builder

File: lib/main.dart [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+  const App({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      debugShowCheckedModeBanner: false,
+      theme: ThemeData(useMaterial3: true),
+      home: Scaffold(
+        appBar: AppBar(
+          title: const Text('Flutter Internals'),
```
Comment (KaustubhDoshaheen) [inline lib/main.dart]: same

File: lib/todo_item.dart [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+    if (priority == Priority.urgent) {
+      icon = Icons.notifications_active;
+    }
+
+    if (priority == Priority.normal) {
+      icon = Icons.list;
+    }
+
+    return Padding(
+      padding: const EdgeInsets.all(8),
```
Comment (KaustubhDoshaheen) [inline lib/todo_item.dart:21]: same

File: lib/todo_item.dart [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+      child: Row(
+        children: [
+          Icon(icon),
+          const SizedBox(width: 12),
```
Comment (KaustubhDoshaheen) [inline lib/todo_item.dart:25]: same

**Review by KaustubhDoshaheen — CHANGES_REQUESTED** [review]

File: lib/keys.dart (hunk 4 above)
[the remark this answers is not in the document]
Comment (richajituri1) [reply lib/keys.dart:63]: resolved

File: lib/keys.dart (hunk 5 above)
Comment (richajituri1) [reply lib/keys.dart]: resolved

File: lib/keys.dart (hunk 6 above)
Comment (richajituri1) [reply lib/keys.dart]: resolved

File: lib/keys.dart (hunk 7 above)
Comment (richajituri1) [reply lib/keys.dart]: resolved

File: lib/keys.dart (hunk 8 above)
Comment (richajituri1) [reply lib/keys.dart]: resolved

File: lib/keys.dart (hunk 9 above)
Comment (richajituri1) [reply lib/keys.dart]: resolved

File: lib/keys.dart (hunk 10 above)
Comment (richajituri1) [reply lib/keys.dart]: resolved

File: lib/main.dart (hunk 11 above)
Comment (richajituri1) [reply lib/main.dart]: resolved

File: lib/todo_item.dart (hunk 12 above)
Comment (richajituri1) [reply lib/todo_item.dart:21]: resolved

File: lib/todo_item.dart (hunk 13 above)
Comment (richajituri1) [reply lib/todo_item.dart:25]: resolved

**Pull request merged**